From 043ea8a4831975373e1cb5c63dc09ee5609f5d36 Mon Sep 17 00:00:00 2001
Message-Id: <043ea8a4831975373e1cb5c63dc09ee5609f5d36.1529666554.git.maciej.zenon.borzecki@canonical.com>
From: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
Date: Fri, 22 Jun 2018 12:34:30 +0200
Subject: [PATCH] dirs: improve identification of Arch Linux like systems

Arch Linux derivatives correctly set ID_LIKE=archlinux but we use 'arch' in our
detection (which is only set by Arch as ID=arch).

Signed-off-by: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
---
 dirs/dirs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dirs/dirs.go b/dirs/dirs.go
index df4c028083c16c6abfa7438a5b11459e9dcc9b07..2e84ae962665f97e1bc2312993a89b1a8bc58bbf 100644
--- a/dirs/dirs.go
+++ b/dirs/dirs.go
@@ -178,7 +178,7 @@ func SetRootDir(rootdir string) {
 	}
 	GlobalRootDir = rootdir
 
-	if release.DistroLike("fedora", "arch", "manjaro") {
+	if release.DistroLike("fedora", "archlinux", "manjaro") {
 		SnapMountDir = filepath.Join(rootdir, "/var/lib/snapd/snap")
 	} else {
 		SnapMountDir = filepath.Join(rootdir, defaultSnapMountDir)
-- 
2.17.1

